lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -70,6 +76,62 @@ defmodule Archethic.Mining.PendingTransactionValidation do
     end
   end
 
+  @spec validate_previous_transaction_type?(Archethic.TransactionChain.Transaction.t()) ::
+          boolean
+  def validate_previous_transaction_type?(tx) do
+    case Transaction.network_type?(tx.type) && tx.type not in [:code_proposal] do
+      false ->
+        #  not a network tx , no need to validate with
+        #  last tx
+        true
+
+      true ->
+        # when network tx , check with previous transaction
+        do_validate_previous_transaction_type?(tx)
+    end
+  end
+
+  @spec do_validate_previous_transaction_type?(Archethic.TransactionChain.Transaction.t()) ::
+          boolean
+  def do_validate_previous_transaction_type?(curr_txn = %Transaction{}) do
+    prev_address = Transaction.previous_address(curr_txn)
+
+    case TransactionChain.get_transaction(prev_address, [:address, :type]) do
+      {:error, :transaction_not_exists} ->
+        # previous txn do not exist, a genesis_txn
+        # TODO imp:
+        # impl to different b/w user generated netowrk_txn
+        true
+
+      {:error, :invalid_transaction} ->
+        # Last txn is
+        # TODO imp: impl handling of last KO txn
+        false
+
+      {:ok, last_txn = %Transaction{}} ->
+        match_last_tx_type?(curr_txn.type, last_txn.type) &&
+          Transaction.verify_previous_signature?(curr_txn)
+    end
+  end
+
+  @spec match_last_tx_type?(:atom, :atom) :: boolean
+  def match_last_tx_type?(curr_tx_type, last_txn_type) do
```

Comment:
The logic is here but using pattern matching would be a bit clearer in my opinion